src/components/PostDetails.tsx

```diff
@@ -2,105 +1,139 @@
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { addComment, deleteComment, getCommentsByPost } from '../api/comments';
+
+type Props = {
+  post: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isAddingComment, setIsAddingComment] = useState(false);
```

Comment:
Instead of `isAddingComment`, consider using `isCommentBeingAdded` for better code readability.